src/main/java/org/commcare/modern/util/PerformanceTuningUtil.java

```diff
@@ -0,0 +1,66 @@
+package org.commcare.modern.util;
+
+/**
+ * A catch-all for centralizing hooks and state for tuning on performance optimizations which may
+ * need to shift constants or boundaries based on the current platform
+ *
+ * Created by ctsims on 6/21/2017.
+ */
+
+public class PerformanceTuningUtil {
+    private static int MAX_PREFETCH_CASE_BLOCK = -1;
+
+    private static final long MB_64 = 64 * 1024 * 2014;
+    private static final long MB_256 = 256 * 1024 * 2014;
+    private static final long MB_1024 = 1024 * 1024 * 2014;
+
+    /**
+     * Update the constant size of cases eligible for batch "pre-fetch" in db ops. This
+     * is roughly the number of cases which can be expected to reliably fit into memory without
+     * lowering the available heap sufficiently to introduce stability concerns.
+     */
+    public static void updateMaxPrefetchCaseBlock(int newMaxSize) {
+        MAX_PREFETCH_CASE_BLOCK = newMaxSize;
+    }
+
+    /**
+     * @return A heuristic for the largest safe value for block prefetch based on the
+     * current device runtime.
+     */
+    public static int identifyDefaultPrefetchBlockSize() {
+        Runtime rt = Runtime.getRuntime();
+        long maxMemory = rt.maxMemory();
+
+        return identifyDefaultPrefetchBlockSize(maxMemory);
+    }
+
+    /**
+     * @return A heuristic for the largest safe value for block prefetch based on a provided
+     * amount of memory which should be available for optimizations.
+     */
+    public static int identifyDefaultPrefetchBlockSize(long availableMemoryInBytes) {
+        if (availableMemoryInBytes == 0 || availableMemoryInBytes == -1) {
```

Comment:
Is this just if the device doesn't support getting the size of memory? How did you decide what to return in this case?